core/src/main/java/org/jruby/javasupport/ext/JavaLang.java

```diff
@@ -340,6 +344,26 @@ public static IRubyObject integer_p(final ThreadContext context, final IRubyObje
                                                     val instanceof java.math.BigInteger);
         }
 
+        @JRubyMethod(name = "coerce")
+        public static IRubyObject coerce(final ThreadContext context, final IRubyObject self, final IRubyObject type) {
+            java.lang.Number val = (java.lang.Number) self.toJava(java.lang.Number.class);
+
+            // NOTE: a basic stub that always coverts Java numbers to Ruby ones (for simplicity)
+            // gist being this is not expected to be used heavily, if so should get special care
+            final IRubyObject value;
+            if (val instanceof java.math.BigDecimal) {
+                final RubyClass klass = context.runtime.getClass("BigDecimal");
+                if (klass == null) { // user should require 'bigdecimal'
```

Comment:
guess its not that common to use - like in Java, `BigInteger` is probably used more than BigDecimal.
... but you're right if all others are core maybe this shouldn't be a lib.
was also thinking of `BigDecimal` auto-converting Java<->Ruby, however more tricky the library isn't loaded.